### PR TITLE
Fix injecting volumes into rollover jobs

### LINKED ISSUE #434: jaeger services failed to deploy, if "es.use-aliases" is enabled
I enabled rollover support in the CR file, jaeger services are failed to deploy.
jaeger-operator=1.12.0

Error in the jaeger-operator log:
```
time="2019-05-23T13:51:36Z" level=error msg="failed to apply the changes" error="CronJob.batch \"jaegerqe-ro-es-rollover\" is invalid: spec.jobTemplate.spec.template.spec.containers[0].volumeMounts[0].name: Not found: \"certs\"" execution="2019-05-23 13:51:36.720354103 +0000 UTC" instance=jaegerqe-ro namespace=jkandasa
```

CR file:
```yaml
apiVersion: jaegertracing.io/v1
kind: Jaeger
metadata:
  name: jaegerqe-ro
spec:
  ingress:
    security: none
  strategy: production
  collector:
    replicas: 1
    image: jaegertracing/jaeger-collector:1.12
    resources:
      requests:
        cpu: "1"
        memory: "512Mi"
      limits:
        cpu: "1"
        memory: "512Mi"
    options:
      log-level: info
      metrics-backend: prometheus
      collector:
        num-workers: 20
        queue-size: 100000
      es:
        bulk:
          size: 524288
          workers: 3
          flush-interval: 1000ms
        tags-as-fields:
          all: false
  query:
    replicas: 1
    image: jaegertracing/jaeger-query:1.12
    resources:
      requests:
        cpu: "500m"
        memory: "512Mi"
      limits:
        cpu: "500m"
        memory: "512Mi"
    options:
      log-level: info
      metrics-backend: prometheus
      query:
        port: 16686
  agent:
    strategy: sidecar
    image: jaegertracing/jaeger-agent:1.12
    resources:
    options:
      log-level: info
      metrics-backend: prometheus
      processor:
        jaeger-compact:
          server-queue-size: 100000
          workers: 1000
  storage:
    type: elasticsearch
    esIndexCleaner:
      enabled: false
    dependencies:
      enabled: false
    elasticsearch:
      nodeCount: 3
      resources:
    options:
      es:
        use-aliases: true
```

operator-log: [jaeger-operator-78b4c8d5dd-bngtl.log](https://github.com/jaegertracing/jaeger-operator/files/3212453/jaeger-operator-78b4c8d5dd-bngtl.log)
rollover-create-mapping log: [jaegerqe-ro-es-rollover-create-mapping-kcdfx.log](https://github.com/jaegertracing/jaeger-operator/files/3212457/jaegerqe-ro-es-rollover-create-mapping-kcdfx.log)

indices in ES cluster:
```
sh-4.2$ curl --cacert /etc/openshift/elasticsearch/secret/admin-ca --key /etc/openshift/elasticsearch/secret/admin-key --cert /etc/openshift/elasticsearch/secret/admin-cert https://elasticsearch:9200/_cat/indices?v
health status index                 uuid                   pri rep docs.count docs.deleted store.size pri.store.size
green  open   jaeger-span-000001    aTZjh2bORzOTRgn32qT-uw   3   1          0            0       972b           486b
green  open   .searchguard          tMIWfKm3SWe_VwO8Znefug   1   1          5            0     71.8kb         35.9kb
green  open   jaeger-service-000001 Cxwx1xjQRN6oxa67eREDAg   3   1          0            0       972b           486b
```

image

### PR DESCRIPTION
**Pull request by pavolloffay**
Resolves #434 

Signed-off-by: Pavol Loffay <ploffay@redhat.com>

### CODE REVIEW

**Comment by pavolloffay** [comment]
I think this deserves a test

**Comment by objectiser** [comment]
@pavolloffay Don't understand how this changes any behaviour, as appending an empty array should have no effect anyway? And not sure how this would fix the referenced problem?

**Comment by pavolloffay** [comment]
This is how pointers work in golang. The items have to be added to the array after they are modified. Or we could use an array of references instead.

File: pkg/strategy/production.go [hunk 1]
```diff
@@ -125,6 +124,9 @@ func newProductionStrategy(jaeger *v1.Jaeger, existingSecrets []corev1.Secret) S
 	if indexCleaner != nil {
 		c.cronJobs = append(c.cronJobs, *indexCleaner)
 	}
+	if len(esRollover) > 0 {
+		c.cronJobs = append(c.cronJobs, esRollover...)
```
Comment (pavolloffay) [inline pkg/strategy/production.go]: @objectiser to help you understand the code. 

The `esRollover` arrays changes made on line 115 are not reflected if the `esRollover` array has been added to `c.cronJobs` before the modification.

Comment (objectiser) [reply pkg/strategy/production.go]: Yes, sorry - only looked at the changes, not the full code.

**Comment by objectiser** [comment]
@pavolloffay agree needs a test, but otherwise looks good.

File: pkg/strategy/production_test.go [hunk 2]
```diff
@@ -164,23 +164,53 @@ func assertDeploymentsAndServicesForProduction(t *testing.T, name string, s S, h
...
 	for _, dep := range c.Deployments() {
 		if strings.HasSuffix(dep.Name, "-query") {
 			assert.Equal(t, 2, len(dep.Spec.Template.Spec.Containers))
 			assert.Equal(t, "jaeger-agent", dep.Spec.Template.Spec.Containers[1].Name)
 		}
 	}
 }
+
+func TestInjectElasticsearch(t *testing.T) {
+	j := v1.NewJaeger("TestAgentSidecarIsInjectedIntoQueryForStreamingForProduction")
```
Comment (objectiser) [inline pkg/strategy/production_test.go]: nit: Should use test name `TestInjectElasticsearch`

**Pull request merged**